proto/service.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+package main;
+option go_package = "github.com/SandyWalsh/runner/gen";
+
+message AuthRequest {
+    string username = 1;
+}
+
+message LaunchRequest {
+    string sessionID = 1;
+    string cmd = 2;
+    repeated string arg = 3;
+}
+
+message NoArg {};
+
+message Status {
+    enum StatusEnum {
+        ERROR = 0;
+        RUNNING = 1;
+        COMPLETED = 2;
+    }
+    StatusEnum status = 1;
+    int32 ExitCode = 2;
+}
+
+enum Error {
+    NOERROR = 0;
+    AUTHERROR = 1;   
+}
+
+message Output {
+    string line = 1;
```

Comment:
A proto3 string must always contain UTF-8 encoded or 7-bit ASCII text as per

https://developers.google.com/protocol-buffers/docs/proto3#scalar

Do we need to enforce this restriction, or can we support arbitrary bytes instead?

Reply:
I can probably do raw bytes. I'll play around with that.

Reply:
While you're here, I see that the field is called `line`, which implies to me that you'll be looking for some sort of `\n` byte before you send any data. What happens if the process never outputs a `\n`? Will you just keep buffering up a really long line and never send it to the stream? Can we design the streaming API such that it doesn't depend on a particular character or byte sequence in order to emit data?

Reply:
no, I was just going to use a `bytes.Buffer` and grab chunks periodically. If they contain a `\n` so be it :)

Reply:
Great! In that case, maybe we should just rename the field to something more generic like `data`.

Reply:
will do ... thanks Zac